### PR TITLE
Implement schemadb

### PR DESCRIPTION
**Pull request by preston-evans98**
SchemaDb is a thin wrapper around RocksDB that allows typed values to be persisted using the sovereign SDK's `Schema` trait.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+authors = ["Sovereign Labs <info@sovereign.xyz>"]
+homepage = "sovereign.xyz"
+publish = false
+repository = "https://github.com/sovereign-labs/sov-db"
+rust-version = "1.67"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[workspace.dependencies]
+# Maintained by soveregin labs
```
Comment (cemozerr) [inline Cargo.toml:24]:
```suggestion
# Maintained by Sovereign Labs
```

File: Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+homepage = "sovereign.xyz"
+publish = false
+repository = "https://github.com/sovereign-labs/sov-db"
+rust-version = "1.67"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[workspace.dependencies]
+# Maintained by Sovereign Labs
+sovereign-sdk = { path = "../sovereign-sdk" }
```
Comment (bkolad) [inline Cargo.toml:25]: Let's change it to git dependency.

File: schemadb/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,396 @@
...
+                    WriteOp::Deletion { key } => db_batch.delete_cf(cf_handle, key),
+                }
+            }
+        }
+        let serialized_size = db_batch.size_in_bytes();
+
+        self.inner.write_opt(db_batch, &default_write_options())?;
+
+        // Bump counters only after DB write succeeds.
+        for (cf_name, rows) in rows_locked.iter() {
```
Comment (bkolad) [inline schemadb/src/lib.rs:247]: Can we combine this loop with the previous one (line 233)? The `for` condition and `match` look the same in both loops.

**Review by bkolad — APPROVED** [review]
LGTM

File: schemadb/src/lib.rs (hunk 3 above)
Comment (preston-evans98) [reply schemadb/src/lib.rs:247]: We can, but it might cause metrics to report incorrectly if the actual write on line 244 fails

Comment (bkolad) [reply schemadb/src/lib.rs:247]: I am ok with the two loops.

File: Cargo.toml (hunk 2 above)
Comment (preston-evans98) [reply Cargo.toml]: Done

**Pull request merged**